core/configuredivecomputerthreads.cpp

```diff
@@ -173,7 +173,7 @@ static dc_status_t read_suunto_vyper_settings(dc_device_t *device, DeviceDetails
 	if (rc != DC_STATUS_SUCCESS)
 		return rc;
 	// in ft * 128.0
-	int depth = feet_to_mm(data[0] << 8 ^ data[1]) / 128;
+	int depth = (int)feet_to_mm(data[0] << 8 ^ data[1]) / 128;
```

Comment:
this is odd... I wonder if `feet_to_mm()` should return an int instead... actually, this might be one of those "Windows is still a 32bit binary" issues. But no, that shouldn't be a problem here... definitely worth looking into

Reply:
> core/units.h:172:static inline unsigned long feet_to_mm(double feet)

`long` and `int` would be the same for `-m32` or around 4.3 million meters of depth, so `unsigned int` is fine too.

Reply:
also, all callers of `feet_to_mm()` should use a `unsigned` receiving variable to avoid possible signed / unsigned warnings. AFAIK, we don't handle negative depth or somehow elevation above sea level.

Reply:
I'm don't know the context of these functions, but relative changes might very well be negative. A unit-conversion function taking a `double` and returning an `unsigned long` does seem a bit surprising.